ports/mimxrt/modpyb.c

```diff
@@ -0,0 +1,41 @@
+/*
+ * This file is part of the MicroPython project, http://micropython.org/
+ *
+ * The MIT License (MIT)
+ *
+ * Copyright (c) 2019 Damien P. George
+ * Copyright (c) 2020 Jim Mussared
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in
+ * all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+ * THE SOFTWARE.
+ */
+
+#include "py/runtime.h"
+#include "led.h"
+
+STATIC const mp_rom_map_elem_t pyb_module_globals_table[] = {
```

Comment:
I'm not 100% sure that this should be in the "pyb" module. Some discussion probably required, but I see the pyb module as primarily "for the pyboard", and for historical reasons, also "STM32 functionality that hasn't made it to the machine module yet".

Unfortunately there isn't really a good precedent here, but I think this is a good an opportunity as ever to make a decision... LED is kind of a funny one...it's somewhere between machine and "board".

Perhaps we could defer this decision by just exposing the Pin object for now (i.e. machine.Pin), just providing the high/low/value methods.

@dpgeorge ?

Reply:
I agree, there should not be a `pyb` module in this port.

The nrf port has a `board` module with `board.LED`.  And `examples/hwapi/` puts `LED` configuration in a Python `hwconfig` module.  But regardless of whether it's implemented in C or Python, the important decision is where to put LED, and whether it's just an alias of Pin/Signal, or is a specific implementation.

> Perhaps we could defer this decision by just exposing the Pin object for now (i.e. machine.Pin), just providing the high/low/value methods.

Yes, that would be the easy path.  Although it is nice to have pre-defined LEDs on the correct pins so you don't have to look it up in the board reference.

Reply:
I guess I could put it in the ```machine``` module then in addition to a future ```pin``` object. 
My take was more like looking at the STM32 port to allow for code reuse. Hence I put it into the ``` pyb``` module. But of course it makes not much sense to have this module in that particular port.

Reply:
done - moved to ```machine``` module and removed ```modpyb``` and any references to ```pyb``` from port